From d5ad9fa976f0ca794a8209d03e254c93d7a939a8 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Tue, 5 Apr 2011 17:59:10 +0200
Subject: [PATCH] patch: hstack

Squashed commit of the following:

commit 4439c947c25ead3073fc3cb00b03fbfc0914e81a
Author: Mukund Sivaraman <muks@banu.com>
Date:   Tue Apr 5 00:58:50 2011 +0530

    Bug #633969: Check if aux_rect is NULL before dereferencing
    (cherry picked from commit 742446507e819755bd3f44e2d47b157f8cd19ab2)
---
 operations/workshop/hstack.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/operations/workshop/hstack.c b/operations/workshop/hstack.c
index a8a2f59..084336c 100644
--- a/operations/workshop/hstack.c
+++ b/operations/workshop/hstack.c
@@ -46,12 +46,12 @@ get_bounding_box (GeglOperation *operation)
   GeglRectangle *aux_rect = gegl_operation_source_get_bounding_box (operation,
                                                                       "aux");
 
-  if (!in_rect)
+  if (!in_rect || !aux_rect)
     return result;
 
   result = *in_rect;
   if (result.width  != 0 &&
-      result.height  != 0)
+      result.height != 0)
     {
       result.width += aux_rect->width;
       if (aux_rect->height > result.height)
-- 
1.7.4.2

